From 35be8f0377eebe2709e4011010dcb35ea0315ab3 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Thu, 15 May 2014 17:49:22 +0100
Subject: [PATCH 244/351] ASoC: florida: Correct unmasking of DRC interrupt
 for DSP core

We were actually masking the DRC IRQ rather then unmasking it. How this
system was working in this case I can't quite explain but we are
definitely supposed to be unmasking it.

Change-Id: Id724025a89bce27f3e70c0dbdb17378f7859b7e9
Reported-by: Nikesh Oswal <nikesh.oswal@wolfsonmicro.com>
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
---
 sound/soc/codecs/florida.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/florida.c b/sound/soc/codecs/florida.c
index fe79f71..70c8ce0 100644
--- a/sound/soc/codecs/florida.c
+++ b/sound/soc/codecs/florida.c
@@ -2045,7 +2045,7 @@ static int florida_codec_probe(struct snd_soc_codec *codec)
 	snd_soc_dapm_enable_pin(&codec->dapm, "DRC2 Signal Activity");
 	ret = regmap_update_bits(arizona->regmap, ARIZONA_IRQ2_STATUS_3_MASK,
 				 ARIZONA_IM_DRC2_SIG_DET_EINT2,
-				 ARIZONA_IM_DRC2_SIG_DET_EINT2);
+				 0);
 	if (ret != 0) {
 		dev_err(arizona->dev,
 			"Failed to unmask DRC2 IRQ for DSP: %d\n",
@@ -2065,7 +2065,7 @@ static int florida_codec_remove(struct snd_soc_codec *codec)
 	arizona_free_irq(arizona, ARIZONA_IRQ_DSP_IRQ1, priv);
 	regmap_update_bits(arizona->regmap, ARIZONA_IRQ2_STATUS_3_MASK,
 			   ARIZONA_IM_DRC2_SIG_DET_EINT2,
-			   0);
+			   ARIZONA_IM_DRC2_SIG_DET_EINT2);
 
 	priv->core.arizona->dapm = NULL;
 
-- 
1.7.9.5

